src/main/java/com/gildedrose/AgedBrie.java

```diff
@@ -0,0 +1,26 @@
+package com.gildedrose;
+
+public class AgedBrie extends DefaultItem {
```

Comment:
Question: Since `Aged Brie` and `Backstage passes` age the same way, they would both extend the same class?

Reply:
Question: Is the idea to have `DefaultItem`, `legendaryItem` and then each item class will extend those based on how they age?

Reply:
> 

Is this _really_ optimizing too much? When we decided to go down this route my initial reaction was "this is going to be a lot of small classes" but then my head went to, "this early on, it seems smarter to break this apart as much as possible to have a solid foundation to build upon" Since we really don't know what this software is going to become and we inherited this giant block of code I thought it would be best to isolate as much behavior as we can and something as simple as how things increment up and down seemed pretty safe that the behavior of how things appreciate or depreciate will stay pretty consistent but the implementation details of that is what will be most volatile, so with how it is currently set up, the program will be very flexible to those changes.

I would say that a miss was how we implemented the Items. We isolated the behavior of how things update but looking back may have been a little lazy with the items. But, the items are the biggest unknown here since we only have 5-6 in the program and pipeline so who knows what they will look like moving forward. Taking that into consideration maybe there would be a better way to refactor the items to keep them flexible but this early on, with how things are broken apart, it seems easier to put things back together to make changes rather than having to break them apart down the road based on a missed abstraction.

@riccjohn Im basing a lot of this methodology on 99 bottles where Sandi broke the problem down into very isolated pieces almost right off the bat based on the first requirement being introduced.

Reply:
I see. I would be careful when defining relationships this way because then you're opening up the space to have many many sub-classes that might not make sense. Instead of saying this item is a special item and special items do X, Y or Z, perhaps look at it like these are all `Item` and item A can do X, item B can do Y.

Reply:
The removal of the extends and `Updateable` classes in each item is addressed from [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/787edac2b69be3565984ac07747d058edd313cba) to [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/16/commits/b4a7a025d8519c2f1355aff41986085493192ee6) (small commits so the whole solution spanned across multiple)

Would we want to potentially have an `Items` interface with an `age` method for each of the items to implement?